### PR TITLE
CMR 4271

### PR DESCRIPTION
**Pull request by ygliuvt**
This added XML reference response format to variables search and added support of all revisions search to all variable search response formats.

### CODE REVIEW

File: search-app/src/cmr/search/api/variables.clj [hunk 1]
```diff
@@ -56,10 +56,6 @@
   (variable-api-response
    (variable-service/dissociate-variable-to-collections context variable-concept-id body)))
 
-(defn search-for-variables
```
Comment (oubiwann) [inline search-app/src/cmr/search/api/variables.clj]: I'm guessing this wasn't used anywhere? (I don't see a removal of its reference ...)

File: search-app/src/cmr/search/data/query_to_elastic.clj [hunk 2]
```diff
@@ -421,17 +427,22 @@
...
-(defmethod q2e/concept-type->sub-sort-fields :variable
-  [_]
-  [{(q2e/query-field->elastic-field :name :variable) {:order "asc"}}
-   {(q2e/query-field->elastic-field :provider :variable) {:order "asc"}}])
+(defmethod q2e/query->sort-params :variable
+  [query]
+  (let [{:keys [concept-type sort-keys]} query
+        specified-sort (q2e/sort-keys->elastic-sort concept-type sort-keys)
+        default-sort (q2e/sort-keys->elastic-sort concept-type (q/default-sort-keys concept-type))
+        sub-sort-fields (if (:all-revisions? query)
```
Comment (mschmele) [inline search-app/src/cmr/search/data/query_to_elastic.clj]: Nitpick: I think we should avoid having conditions inside `let` blocks. To me, the block below is a little easier to read and refactor in the future if needed

```
(concat
  (or (q2e/sort-keys->elastic-sort concept-type sort-keys)
      (q2e/sort-keys->elastic-sort concept-type (q/default-sort-keys concept-type))
  (if (:all-revisions? query)
    (all-revision-sub-sort-fields :variable)
    variable-latest-sub-sort-fields)))

File: search-app/src/cmr/search/api/variables.clj (hunk 1 above)
Comment (ygliuvt) [reply search-app/src/cmr/search/api/variables.clj]: Yes, this should have been dropped when I moved the search endpoint to routes.clj.

File: search-app/src/cmr/search/data/query_to_elastic.clj (hunk 2 above)
Comment (ygliuvt) [reply search-app/src/cmr/search/data/query_to_elastic.clj]: I am fine with `if` within let block. In this case, it is basically setting a intermediate variable and the variable name (`sub-sort-fields` makes the code more readable). The combined way is harder to read in my opinion.

**Pull request merged**